admin/template_select.php

```diff
@@ -188,8 +188,42 @@
                  $contents[] = ['text' => TEXT_INFO_TEMPLATE_AUTHOR . $template_info[$tInfo->template_dir]['author']];
                  $contents[] = ['text' => TEXT_INFO_TEMPLATE_VERSION . $template_info[$tInfo->template_dir]['version']];
                  $contents[] = ['text' => TEXT_INFO_TEMPLATE_DESCRIPTION . '<br>' . $template_info[$tInfo->template_dir]['description']];
-                 $contents[] = ['align' => 'text-center',
-                    'text' => '<a href="' . zen_href_link(FILENAME_TEMPLATE_SELECT, 'page=' . $_GET['page'] . '&tID=' . $tInfo->template_id . '&action=edit') . '" class="btn btn-primary" role="button">' . TEXT_INFO_EDIT_INTRO . '</a>' .
+                 if ($template_info[$tInfo->template_dir]['has_template_settings'] === true) {
+                     $template_settings_button =
+                        '<button type="button" class="btn btn-info" data-toggle="modal" data-target="#view-settings">' .
+                            TEXT_VIEW_TEMPLATE_SETTINGS .
+                        '</button> ';
+                     $template_settings = file_get_contents($template_info[$tInfo->template_dir]['template_path'] . '/template_settings.php');
```

Comment:
Is there any chance of this coming back as boolean `false`? (eg: fail to open/load file?)

Reply:
I guess there's a possibility; the update to the template-function checks that the file exists but it could be unreadable.